canonical/kfp-operators
# `kfp-operators` upgrades from 2.0.0-alpha.7 -> 2.0.3 for CKF 1.8 release

This PR merges `1.8-updates-dev-branch` into `main`, which includes changes for (but not limited to):
* Updating the kfp-operators from 2.0.0-alpha.7 to 2.0.3 https://github.com/canonical/kfp-operators/pull/331 and https://github.com/canonical/kfp-operators/pull/379
* Ensuring the Services of the kfp-operators are correctly configured https://github.com/canonical/kfp-operators/pull/361, https://github.com/canonical/kfp-operators/pull/358
* Configuring the kfp-persistence charm with a SA token https://github.com/canonical/kfp-operators/pull/349
* Testing v1 and v2 pipelines on the CI https://github.com/canonical/kfp-operators/pull/383

### Testing instructions

Tesnting environment:
* Microk8s 1.25-strict/stable
> Make sure you run `microk8s config > ~/.kube/config` before continuing. Some kfp-operators components
use in cluster kubeconfig.
 
* juju 3.1/stable

1. Deploy the bundle provided in this PR `juju deploy ./ckf-1.8-dev.yaml --trust`
2. Refresh all charms to the latest upload of this PR

```
juju refresh kfp-api                    --channel latest/edge/pr-362 --resource oci-image=gcr.io/ml-pipeline/api-server:2.0.3
juju refresh kfp-db                     --channel 8.0/stable
juju refresh kfp-persistence            --channel latest/edge/pr-362 --resource oci-image=gcr.io/ml-pipeline/persistenceagent:2.0.3
juju refresh kfp-profile-controller     --channel latest/edge/pr-362 --resource oci-image=python:3.7
juju refresh kfp-schedwf                --channel latest/edge/pr-362 --resource oci-image=gcr.io/ml-pipeline/scheduledworkflow:2.0.3
juju refresh kfp-ui                     --channel latest/edge/pr-362 --resource ml-pipeline-ui=gcr.io/ml-pipeline/frontend:2.0.3
juju refresh kfp-viewer                 --channel latest/edge/pr-362 --resource kfp-viewer-image=gcr.io/ml-pipeline/viewer-crd-controller:2.0.3
juju refresh kfp-viz                    --channel latest/edge/pr-362 --resource oci-image=gcr.io/ml-pipeline/visualization-server:2.0.3
```

3. Ensure the `apiserver` is up and ready

```
$ kubectl logs -nkubeflow kfp-api-0  -c apiserver -f

# The last message from the apiserver should be something like this:
2023-11-17T08:42:23.667Z [pebble] Service "apiserver" starting: bash -c 'sleep 1.1 && /bin/apiserver --config=/config --sampleconfig=/config/sample_config.json -logtostderr=true '
2023-11-17T08:42:24.811Z [apiserver] I1117 08:42:24.811447     104 client_manager.go:170] Initializing client manager
2023-11-17T08:42:24.811Z [apiserver] I1117 08:42:24.811679     104 config.go:57] Config DBConfig.MySQLConfig.ExtraParams not specified, skipping
2023-11-17T08:42:25.036Z [apiserver] I1117 08:42:25.036461     104 client_manager.go:511] Successfully created bucket mlpipeline
2023-11-17T08:42:25.036Z [apiserver] I1117 08:42:25.036823     104 swf.go:64] (Expected when in cluster) Failed to create scheduled workflow client by out of cluster kubeconfig. Error: stat /root/.kube/config: no such file or directory
2023-11-17T08:42:25.036Z [apiserver] I1117 08:42:25.036853     104 swf.go:66] Starting to create scheduled workflow client by in cluster config.
2023-11-17T08:42:25.037Z [apiserver] I1117 08:42:25.037592     104 client_manager.go:214] Client manager initialized successfully
2023-11-17T08:42:27.138Z [apiserver] I1117 08:42:27.138649     104 main.go:284] All samples are loaded.
2023-11-17T08:42:27.138Z [apiserver] I1117 08:42:27.138709     104 main.go:143] Starting Http Proxy
2023-11-17T08:42:27.139Z [apiserver] I1117 08:42:27.139794     104 main.go:96] Starting RPC server 
```

5. Login into the Kubeflow dashboard
6. Create an experiment
7. Create a run from the "Data passing..." sample pipeline.
8. Wait for completion
9. Create a recurring run and wait for completion

Please note both CI tests and manual test should pass before we merge this PR.

##### Test bundle

[ckf-1.8-dev.txt](https://github.com/canonical/kfp-operators/files/13416449/ckf-1.8-dev.txt)
